Cookbook-ios/CodeScanner/CodeScannerController.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  CodeScannerController.swift
+//  Cookbook-ios
+//
+//  Created by Carolina Arcos on 2/11/20.
+//  Copyright © 2020 Wolox. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+class CodeScannerController: GeneralViewController, AVCaptureMetadataOutputObjectsDelegate {
+    // MARK: Properties
+    @IBOutlet weak var cameraView: UIView!
+    private lazy var _scanner: Scanner? = Scanner(withViewController: self,
+                                                  view: cameraView,
+                                                  codeOutputHandler: handleCode)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        feature = Features.codeScanner
+        configureNavigationBar()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        validateCameraAccess()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        stopScanner()
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateCameraLayer()
+    }
+}
+
+// MARK: - Camera access
+private extension CodeScannerController {
+    func validateCameraAccess() {
+        guard UIImagePickerController.isSourceTypeAvailable(.camera) else {
+            showErrorAlert(message: "You don't have a camera")
+            return
+        }
+        
+        AVCaptureDevice.requestAccess(for: AVMediaType.video) { [unowned self] hasAccess in
+            guard hasAccess else {
+                self.showErrorAlert(message: "Your camera access was rejected. Please change this configuration in Settings")
+                return
+            }
+            
+            self.startScanner()
+        }
+    }
+    
+    func showErrorAlert(message: String) {
+        let alert = UIAlertController(title: "No camera access",
```

Comment:
Is it worth it to include localized strings?

Reply:
I'm going to add the file with string constants